### DISCUSSION TITLE
Slow degrading performance, again?

### DISCUSSION THREAD

**Discussion by n123456789k** [community/community#7920] [Codespaces]
Over the past week I've had:
- Slow browser based vscode loading
- Failing to load in general
- Not showing the ports tab
- Not showing the git tab
- Failing to load documents

This is on the base container image with nothing running.

I've ended up re-building like 10x codespaces to try to fix the issue. Contact Github support with no fix.

What's going on? I understand this is in Beta but constant "liveshare" errors and timing out in the dev console request logs for Chrome.

It's 50/50, I wait for the container to stop (with timeout) then restart, it ends up like this or.. it sometimes works? However if I'm coding I can't just stop and spin up another codespace as my persisted files are stored to that codespace.

This is literally the base image (no devcontainer), top spec codespace machine type:

![Screenshot 2021-11-19 at 18 57 58](https://user-images.githubusercontent.com/32820112/142676671-211247f4-e327-440d-a7e9-3e43754eef26.png)

**Comment by asciimike** [comment]
Hey folks, sorry you're having issues with Codespaces performance. We're actively investigating these issues and will get them resolved as quickly as possible!

**Reply by n123456789k** [reply]
@asciimike Thanks, do we have an update because this is becoming really annoying that we can't connect half the time and isn't really productive. It's even a simple new codespace we boot up we get the issue. I'm happy to even screenshare this on a call or something.

**Reply by Adam-Pond** [reply]
Would be good to get an update on this. I'm trying codespaces for the first time and the performance is terrible. Trying to work out if this is a transient issue or a systemic blocking issue.

I'm running the karma server, and half the time it won't even start up. When it is running, it'll be working fine (abliet very slow) and then will just stop responding. Undoubtedly, this is a server timeout issue, but it's currently set to 2000ms, and that "should" be enough, even on a slow system.

``` Disconnected reconnect failed before timeout of 2000ms (server shutting down)```
